test/test_grade.py

```diff
@@ -205,3 +206,24 @@ def test_notebooks_with_pdfs(expected_points):
         [os.path.splitext(f)[0] for f in os.listdir("test/submission_pdfs") if not (os.path.isdir(os.path.join("test/submission_pdfs", f)))],
     )
     assert sorted(dir1_contents) == sorted(dir2_contents), f"'{FILE_MANAGER.get_path('notebooks/')}' and 'test/submission_pdfs' have different contents"
+
+@pytest.mark.slow
+@pytest.mark.docker
+def test_single_notebook_grade(expected_points):
```

Comment:
You could make it so this test doesn't require Docker by mocking `otter.grade.launch_grade` and asserting that it gets called with the correct arguments and that its return value is correct.

Reply:
@chrispyles I may be off here so please send me down the right track but I don't think it makes sense to mock the launch_grade function here. I don't change any parameters sent to launch_grade when there is a single notebook in the path. I just strip off the notebook name, update the ext field appropriately and give launch_grade what it expects -- a directory path, .zip etc.

Does that make sense?

Reply:
@sean-morris So we already have tests that check that when `launch_grade` is passed these parameters it does what it's supposed to, meaning that we already know without this additional test that `launch_grade` works as expected. This test isn't adding anything by running `launch_grade` but just slowing the tests down since we need to build another Docker image and run a container again. So instead, we can mock `launch_grade` and verify that it gets passed the expected parameters (since we already know it works) and check that the return value of `launch_grade` is used correctly.